File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        if(nums.size() == 1){
            return nums[0];
        }
        int candidate = nums[0], count = 1;
        for(int i = 1; i < nums.size(); ++i){
            count+=(nums[i] == candidate) ? 1 : -1;
            if(count == 0){
                candidate = nums[i];
                count = 1;
            }
        }
        return candidate;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def recur(self, nums: List[int], idx: int, candidate: List[int], ans: List[List[int]]) -> None:
        if idx == len(nums):
            ans.append(candidate[:])
            return
        candidate.append(nums[idx])
        self.recur(nums, idx + 1, candidate, ans)
        candidate.pop()
        while idx + 1 < len(nums) and nums[idx + 1] == nums[idx]:
            idx+=1
        self.recur(nums, idx + 1, candidate, ans)
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ans = []
        self.recur(nums, 0 , [], ans)
        return ans 